crates/swc_ecma_utils/src/function/fn_env_hoister.rs

```diff
@@ -126,57 +145,378 @@ impl FnEnvState {
+            .computed
+            .get_or_insert_with(|| private_ident!(span, "_superprop_set"))
+            .clone()
+    }
 }
 
-impl<'a> FnEnvHoister<'a> {
-    pub fn new(state: &'a mut FnEnvState) -> Self {
-        FnEnvHoister { state }
+impl Take for FnEnvHoister {
+    fn dummy() -> Self {
+        Self::default()
     }
 }
 
-impl<'a> VisitMut for FnEnvHoister<'a> {
+impl VisitMut for FnEnvHoister {
     noop_visit_mut_type!();
 
     fn visit_mut_expr(&mut self, e: &mut Expr) {
         match e {
             Expr::Ident(id) if id.sym == js_word!("arguments") => {
                 let arguments = self
-                    .state
                     .args
                     .get_or_insert_with(|| private_ident!("_arguments"));
                 *e = Expr::Ident(arguments.clone());
             }
             Expr::This(..) => {
-                let this = self
-                    .state
-                    .this
-                    .get_or_insert_with(|| private_ident!("_this"));
-                *e = Expr::Ident(this.clone());
+                let this = self.get_this();
+                *e = Expr::Ident(this);
             }
             Expr::MetaProp(MetaPropExpr {
                 kind: MetaPropKind::NewTarget,
                 ..
             }) => {
                 let target = self
-                    .state
                     .new_target
                     .get_or_insert_with(|| private_ident!("_newtarget"));
                 *e = Expr::Ident(target.clone());
             }
+            // super.foo = 123 => super_get_foo = (value) => super.foo = value
+            Expr::Assign(AssignExpr {
+                left,
+                right,
+                span,
+                op,
+            }) => {
+                let expr = match left {
+                    PatOrExpr::Expr(e) => e,
+                    PatOrExpr::Pat(p) => {
+                        if let Pat::Expr(e) = &mut **p {
+                            e
+                        } else {
+                            e.visit_mut_children_with(self);
+                            return;
+                        }
+                    }
+                };
+                if let Expr::SuperProp(super_prop) = &mut **expr {
+                    let left_span = super_prop.span;
+                    match &mut super_prop.prop {
+                        SuperProp::Computed(c) => {
+                            let callee = self.super_set_computed(left_span);
+
+                            let op = op.to_update();
+
+                            let args = if let Some(op) = op {
+                                let tmp = private_ident!("tmp");
+                                self.extra_ident.push(tmp.clone());
+                                vec![
+                                    Expr::Assign(AssignExpr {
+                                        span: DUMMY_SP,
+                                        left: PatOrExpr::Pat(Box::new(Pat::Ident(
```

Comment:
Can you change this to `PatOrExpr::Pat(tmp.clone().into())`?

Recently I did some work to ease creation of AST and core transforms are the things which plugin authors will look if they are blocked.